lib/presentation/resources/todo_cards/todo_cards.screen.dart

```diff
@@ -0,0 +1,179 @@
+  const TodoCardsSecreen({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _TodoCardsSecreenState();
+}
+
+class _TodoCardsSecreenState extends State<TodoCardsSecreen> {
+  late TodoCardsCubit _todoCardsCubit;
+
+  @override
+  void initState() {
+    super.initState();
+    initialise();
+  }
+
+  Future<void> initialise() async {
+    if (mounted) {
+      _todoCardsCubit =
+          TodoCardsCubit(todoRepository: getItInstance.get<TodoRepository>());
+
+      _todoCardsCubit.getTodoCards();
+    }
+  }
+
+  @override
+  void didChangeDependencies() {
+    //TODO: Initialize the localisation here
+    super.didChangeDependencies();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          //TODO: add from localisation
+          title: const Text('My TODOs'),
+          actions: [
+            _buildCalendarAppBarButton(),
+          ],
+        ),
+        body: BlocProvider(
+            create: (context) => _todoCardsCubit,
+            child: BlocBuilder<TodoCardsCubit, TodoCardsState>(
+                builder: (context, state) {
+              switch (state.runtimeType) {
+                case GetTodoCardsInProgress:
+                  return const Center(
+                    child: CircularProgressIndicator(),
+                  );
+                case GetTodoCardsSuccess:
+                  final todoCards = (state as GetTodoCardsSuccess).todoCards;
+                  return _buildBody(todoCards ?? []);
+                case GetTodoCardsFailure:
+                  return const Center(
+                    child: Column(
+                      children: [
+                        Icon(Icons.error, size: 50, color: Colors.red),
+                        //TODO: add from localisation
+                        Text('Failed to fetch todo cards'),
+                      ],
+                    ),
+                  );
+              }
+
+              return Container();
+            })),
+        floatingActionButton: FloatingActionButton(
+            onPressed: () {
+              //TODO: build the navigation to next screen
+            },
+            child: const Icon(Icons.add)));
+  }
+
+  Widget _buildCalendarAppBarButton() {
+    return Padding(
+      padding: const EdgeInsets.all(AppPadding.p20),
+      child: InkWell(
+        onTap: () => {
+          //TODO: navigate to calendar screen
+          print('Calendar button tapped'),
+        },
+        child: const Icon(
+          Icons.calendar_month,
+        ),
+      ),
+    );
+  }
+
+  Widget _buildBody(List<TodoModel> todoCards) {
+    return LayoutBuilder(builder: (context, dimens) {
+      return ListView.builder(
+        itemCount: todoCards.length,
+        itemBuilder: (context, index) {
+          return _buildCard(todoCards[index]);
+        },
+      );
+    });
+  }
+
+  Widget _buildCard(TodoModel todoCard) {
```

Comment:
@dilumdesilva this can be extracted as a widget TodoCardWidget.

Reply:
That was the plan. I will do it later as I need to have functionalities like delete, pin to top figure out first and then easier to create call backs to the extracted out class.